File: 344.Reverse_String.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        start = 0
        end = len(s) - 1

        while (start < end):
            tmp = s[start]
            s[start] = s[end]
            s[end] = tmp
            
            start += 1
            end -= 1

File: DSA/344.Reverse_String.cpp
// Solution: #1
class Solution {
public:
    void reverseString(vector<char>& s) {
        reverse(s.begin(), s.end());
    }
};

// Solution: #2
class Solution {
public:
    void reverseString(vector<char>& s) {
        int start = 0, end = s.size() - 1;
        char tmp;
        
        do
        {
            tmp = s[start];
            s[start] = s[end];
            s[end] = tmp;

        }while(++start < --end);

    }
};

// Solution: #3
class Solution {
public:
    void reverseString(vector<char>& s) {
        
        int start = 0, end = s.size() - 1;

        while(start < end){

            swap(s[start++], s[end--]);

        }
    }
};